### PR TITLE
refactor(wallet-mobile): new tx review state

### PR DESCRIPTION
**Pull request by banklesss**
Some state variables need to be stateless

### CODE REVIEW

File: apps/wallet-mobile/src/features/ReviewTx/common/hooks/useFormattedMetadata.tsx [hunk 1]
```diff
@@ -42,6 +42,17 @@ export const formatMetadata = async (
...
+  if (msg.length > 1) {
+    try {
+      return JSON.parse(msg.join(''))
+    } catch {
+      return msg[0]
+    }
+  }
+  return msg[0]
+}
+
```
Comment (banklesss) [inline apps/wallet-mobile/src/features/ReviewTx/common/hooks/useFormattedMetadata.tsx]: included a metadata parser

**Review by michalszorad — APPROVED** [review]

**Comment by banklesss** [comment]
draft to add a little fix

**Comment by banklesss** [comment]
ready for review again

File: apps/wallet-mobile/src/features/ReviewTx/common/hooks/useFormattedMetadata.tsx [hunk 2]
```diff
@@ -42,6 +42,17 @@ export const formatMetadata = async (
   }
 }
 
+const parseMsg = (msg: Array<string>) => {
+  if (msg.length > 1) {
+    try {
+      return JSON.parse(msg.join(''))
+    } catch {
+      return msg[0]
```
Comment (banklesss) [inline apps/wallet-mobile/src/features/ReviewTx/common/hooks/useFormattedMetadata.tsx]:
```suggestion
      return ''
```

File: apps/wallet-mobile/src/features/ReviewTx/common/hooks/useFormattedMetadata.tsx (hunk 1 above)
Comment (banklesss) [reply apps/wallet-mobile/src/features/ReviewTx/common/hooks/useFormattedMetadata.tsx]: outdated

**Pull request merged**